RFC-by-stage/1-approved/separate-components.md

```diff
@@ -0,0 +1,36 @@
+---
+Status: `Proposal` # Please do not change this.
+Implementer: # It will be changed upon merging and as it moves through the RFC stages
+---
+
+# Separate components between React, jQuery and Vanilla
+
+## The issue to be solved
+
+We currently share a lot of code between the vanilla, jQuery and React components.
+This has led to a bunch of issues including the fact that we're not using each framework to its fullest potential.
+
+## A short description of the solution
+
+I propose we create separate entry-points for each language and maintain a separation for the JavaScript used in each.
+We should still re-use the same CSS though or aim to at least use the same token base (which isn't insufficiently separated yet either).
+
+## Technical details
+
+The implementation would be split in separate entry points.
+So inside the monorepo we would have a structure like this:
+
+```sh
+.
+└── packages
+    ├── react
```

Comment:
rather than `packages/buttons/react` ? in your example, are you proposing new npm names?

Reply:
I'm not set on the naming. I proposed this structure because it would make dependency management across the same architecture easier. I would assume with what I propose you would only ever use the react entry point for npm which would remain unchanged. Vanilla or jquery does not use need npm packages and does not have them now.